File: MyWebApp/AcessoDados/Entidades/Produto.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyWebApp.AcessoDados.Entidades
{
    [Table("Produto")]
    public class Produto
    {
        [Key,DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Required]
        [StringLength(255)]
        public string Nome { get; set; }
        [Required]
        public string Descricao { get; set; }
        [Required]
        public decimal Preco { get; set; }
    }
}

File: MyWebApp/wwwroot/ts/produto.ts
export class Produto {
    private _id: number = 0;
    private _nome: string;
    private _descricao: string;
    private _preco: number;
    //constructor(private _descricao: string, private _preco: number) {

    //}

    get id() {
        return this._id;
    }

    set id(id: number) {
        this._id = id;
    }

    get nome() {
        return this._nome;
    }

    set nome(nome: string) {
        this._nome = nome;
    }

    get descricao() {
        return this._descricao;
    }

    set descricao(descricao: string) {
        this._descricao = descricao;
    }

    get preco() {
        return this._preco;
    }

    set preco(preco: number) {
        this._preco = preco;
    }
}
